### PR TITLE
Solved all issues

### PR DESCRIPTION
**Pull request by Arihant-Bhandari**
Solved xor.py's issues with backprop and hyperparameter sensitivity.

Solved dig.py's issue with backprop and forward prop.

### CODE REVIEW

**Comment by Kislay0** [comment]
Updated issue 1 and 2 as well

**Comment by bhqv** [comment]
# Errors:
## dig.py
- Data read incorrectly (dataset inside data/train.csv)
- accuracy received is worse than guessing randomly ~0.09 (after correctly reading dataset)

## xor.py
- Hyperparameters not fixed
